### PR TITLE
POSIX-compliant implementation of relative_path.sh

### PR DESCRIPTION
**Pull request by ghost**
This is an alternative implementation of relative_path.sh, authored by @simonair on http://stackoverflow.com/a/14914070
Known limitations: it doesn't clean multiple slashes and doesn't resolve symlinks.
Let's see if it passes Travis CI.

### CODE REVIEW

File: contrib/relative_path.sh [hunk 1]
```diff
@@ -1,49 +1,29 @@
...
-if [[ -n $result ]] && [[ -n $forward_part ]]; then
-    result="$result$forward_part"
-elif [[ -n $forward_part ]]; then
-    # extra slash removal
-    result="${forward_part:1}"
-fi
-
-echo $result
+#!/bin/sh
+# This file is a part of StackOverflow. License is CC-BY-SA
```
Comment (tkelman) [inline contrib/relative_path.sh]: "based on code from" - and add this file to the exception list in `contrib/add_license_to_files.jl`

File: contrib/relative_path.sh [hunk 2]
```diff
@@ -1,49 +1,29 @@
-#!/usr/bin/env bash
-# This file is a part of Julia. License is MIT: http://julialang.org/license
-
-# both $1 and $2 are absolute paths beginning with /
-# returns relative path to $2/$target from $1/$source
```
Comment (tkelman) [inline contrib/relative_path.sh]: leave this comment in

**Comment by tkelman** [comment]
seems to work on travis. does the existing version handle symlinks or multiple slashes? how does this handle the problem cases from https://github.com/JuliaLang/julia/pull/14097 / #14103?

**Comment by tkelman** [comment]
there's also a comment on SO that "Doesn't work when the first path ends in / I'm afraid." not sure what to make of that or how general we need this to be.

**Comment by albap** [comment]
> there's also a comment on SO that "Doesn't work when the first path ends in / I'm afraid."

I did a quick test and it appears to work. Maybe that comment refers to another implementation, there are 3 of them in simon's answer.
Can you test in your machine too?

About multiple slashes, if it's really a concern, we can pipe the output of relpath through sed s#//*#/#g
But I don't know how to deal with symlinks. But if the current version of relative_path.sh doesn't deal with symlinks, then we're fine to merge.

**Comment by tkelman** [comment]
there were failure cases linked above that we want to be sure we're not reintroducing

**Comment by tkelman** [comment]
If we decide this version is okay it would also be worth asking @simonair if he would be willing to relicense his code snippet as MIT.

**Comment by simonair** [comment]
I'm honored to see my little code snippet be of use. I would be happy to license it under the MIT license – what do I need to do?

**Comment by StefanKarpinski** [comment]
Saying so is all you need to do. Thanks!

**Comment by tkelman** [comment]
Yep, thanks much! We can then remove the license special-casing for that file, and give it the standard same MIT banner we give everything else. More convenient to have fewer exceptions.

File: contrib/add_license_to_files.jl [hunk 3]
```diff
@@ -30,6 +30,7 @@ const excludedirs = [
 
 const skipfiles = [
     "../contrib/add_license_to_files.jl",
+    "../contrib/relative_path.sh",
```
Comment (tkelman) [inline contrib/add_license_to_files.jl]: (no longer needed)

**Comment by albap** [comment]
With licensing solved, we just have to wait for @Keno to give the green-light and it can be merged.

**Comment by albap** [comment]
Two simple questions.

- Do we need to deal with multiple slashes?
If so, the solution is a one line change.
-echo "$relative"
+echo "$relative" | sed s#//*#/#g

- Do we need to handle symlinks?
I don't know. Apparently the current version doesn't?
If so we should be fine to merge.

Can someone answer these questions?

**Comment by Keno** [comment]
Did you check that it correctly handles the two corner cases we had issues for above, i.e.
`/foo/bar` relative to `/foo/bar-baz`? and
`/foo` relative to `/bar`?

**Comment by albap** [comment]
Is this what you ask?
```
$ ./relative_path.sh /foo/bar /foo/bar-baz
../bar-baz
$ ./relative_path.sh /foo /bar
../bar
```

**Comment by Keno** [comment]
Yes, those were the two failure cases in the old script that we had to fix. LGTM then.

**Comment by albap** [comment]
Do you think it's necessary to deal with multiple slashes?
-echo "$relative"
+echo "$relative" | sed s#//*#/#g

Or can it be merged already?

**Comment by tkelman** [comment]
Doesn't look like the current script does so. Everywhere we call it should be getting passed off to things that will deal with multiple slashes as expected.

**Pull request merged**